src/domain/entities/LandingNode.ts

```diff
@@ -92,44 +97,69 @@ export const updateLandingNodes = (
         })
         .compact()
         .value();
+
+    return updatedNodes.map(node => applyFavicon(node));
+};
+
+const applyFavicon = (parent: LandingNode): LandingNode => {
+    const spreadFaviconToChildren = (children: LandingNode[], favicon: string): LandingNode[] => {
+        return _.map(children, child => {
+            return {
+                ...child,
+                favicon: favicon,
+                children: spreadFaviconToChildren(child.children, favicon),
+            };
+        });
+    };
+
+    return {
+        ...parent,
+        children: spreadFaviconToChildren(parent.children, parent.favicon),
```

Comment:
**[Already changed]** Do you see the pattern? indeed the `updatedNodes` where we want to spread the favicon are only the landings themselves. So we can put the functionality of `applyFavicon` inside the `.map` return and simplify the recursive function. Also, as we are executing the recursive function inside another recursive function `updateLandingNodes` updating the children N children times. The best option would be to traverse the tree once applying the favicon simultaneously. But since we are running out of time we can also traverse it twice. Check `updateLandings` L110